### ISSUE TITLE
Each time a new user is created with a number after migration (User1:example.com)

### ISSUE THREAD

**Issue by rtest12** [matrix-org/synapse#12307]
I have the same problem as #9514
I migrated to a new server, no upgrade, just latest version to latest.
Users1-2-3 began to be created
I changed the old value idp_id from 
```
matrix_synapse_configuration_extension_yaml: |
  oidc_providers:
   - idp_id: keycloak
     idp_name: "keycloak"
```
For new
```
matrix_synapse_configuration_extension_yaml: |
  oidc_providers:
   - idp_id: oidc
     idp_name: "keycloak"
```

I restarted the entire installation, but nothing has changed, still when I log in I get "User1:example.com" 
I don't understand what else I missed?

all conf

```
matrix_synapse_configuration_extension_yaml: |
 oidc_providers:
  - idp_id: oidc
    idp_name: "keycloak"
    issuer: "https://id.exam.com/auth/realms/myrealm"
    client_id: "synapse"
    client_secret: "mykey"
    scopes: ["openid", "profile"]
    user_mapping_provider:
      config:
          localpart_template: "{% raw %}{{ user.preferred_username }}{% endraw %}"
          display_name_template: "{% raw %}{{ user.name }}{% endraw %}"
          email_template: "{% raw %}{{ user.email }}{% endraw %}"


```
Also additionally added
`   allow_existing_users: true`
But it didn't help

**Comment by anoadragon453** [comment]
I believe setting `idp_id` to "oidc" is only required if your config previously used the old `oidc_config` configuration section. The old section would implicitly have "oidc" as the the `idp_id`, which is why this is necessary.

It doesn't appear that your old config used the `oidc_config` configuration section, but instead makes use of the modern `oidc_providers` config, with "keycloak" as the `idp_id`. In that case, it makes sense that changing from "keycloak" to "oidc" would result in new users being created.

TL;DR I don't believe changing the value of `idp_id` is necessary. It is only necessary if you were using the old `oidc_config` configuration format, which doesn't appear to be the case for you.

**Comment by rtest12** [comment]
`It doesn't appear that your old config used the oidc_config configuration section, but instead makes use of the modern oidc_providers config, with "keycloak" as the idp_id.`
Yes
` In that case, it makes sense that changing from "keycloak" to "oidc" would result in new users being created.`
Ok, but new users are created even if I don't change anything from the original configuration.
`TL;DR I don't believe changing the value of idp_id is necessary. It is only necessary if you were using the old oidc_config configuration format, which doesn't appear to be the case for you.`
What should I do then?

**Comment by MadLittleMods** [comment]
@rtest12 If you remember the context, it would still be nice to know what your mistake was 🙂

Closing due to now response from reporter (feel free to respond with that context though)
